### PR TITLE
Fixes for issues found during gdi spec review

### PR DESCRIPTION
**Pull request by tsloughter-splunk**
Fixes:

#47 
#46 
#41 
#42 
#43

### CODE REVIEW

**Comment by tsloughter-splunk** [comment]
Added fix for #44

**Comment by tsloughter-splunk** [comment]
Added fix for #49

**Comment by tsloughter-splunk** [comment]
Added fix for #54

**Review by aurbiztondo-splunk — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -166,6 +166,13 @@ To enable instrumentation libraries manually, see [Manual library instrumentatio
 Documentation on how to manually instrument a Ruby application is available in the 
 [OpenTelemetry official documentation](https://opentelemetry.io/docs/instrumentation/ruby/manual/).
 
+To extend the instrumentation with the OpenTelemetry Instrumentation for Ruby,
+you have to use a compatible API version.
+
+The Splunk Distribution of OpenTelemetry Ruby version 0.1.0 is compatible with:
+
+- OpenTelemetry API version 1.0.0
```
Comment (jtmal-signalfx) [inline README.md:174]: Shouldn't we also list supported versions of SDK and instrumentations (`opentelemetry-instrumentation-base`)?

**Review by jtmal-signalfx — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (pellared) [reply README.md:174]: I think at least giving the version of the SDK would be good enough

File: README.md [hunk 2]
```diff
@@ -166,6 +166,13 @@ To enable instrumentation libraries manually, see [Manual library instrumentatio
 Documentation on how to manually instrument a Ruby application is available in the 
 [OpenTelemetry official documentation](https://opentelemetry.io/docs/instrumentation/ruby/manual/).
 
+To extend the instrumentation with the OpenTelemetry Instrumentation for Ruby,
+you have to use a compatible API version.
+
+The Splunk Distribution of OpenTelemetry Ruby version 0.1.0 is compatible with:
```
Comment (pellared) [inline README.md:172]: For easier maintenance, I suggest changing this line to:

```suggestion
The Splunk Distribution of OpenTelemetry Ruby is compatible with:
```

Consider calling out the supported version in the CHANGELOG. For reference:

- https://github.com/signalfx/splunk-otel-java/blob/main/CHANGELOG.md
- https://github.com/signalfx/splunk-otel-go/blob/main/CHANGELOG.md
- https://github.com/signalfx/splunk-otel-python/blob/main/CHANGELOG.md

**Review by pellared — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,4 +7,23 @@ and this repository adheres to [Semantic Versioning](https://semver.org/spec/v2.
...
 ## Unreleased
 
+### Added
 
+- [pre-configured support for Smart
+  Agent](https://github.com/signalfx/splunk-otel-ruby/pull/8)
+- [support for SPLUNK_TRACE_RESPONSE_HEADER_ENABLED environment
+  variable](https://github.com/signalfx/splunk-otel-ruby/pull/38)
+  
+## v0.1.0 - 2022-07-27
```
Comment (pellared) [inline CHANGELOG.md:17]: > MUST state version of OpenTelemetry repository built against if applicable

https://github.com/signalfx/gdi-specification/blob/89371378371723be768e90b6d1beb44fccf42c00/specification/repository.md?plain=1#L130

I know that it is about GH Releases, but often it is the same content as in the CHANGELOG.md.

File: README.md (hunk 2 above)
Comment (tsloughter-splunk) [reply README.md]: Done.

File: CHANGELOG.md (hunk 3 above)
Comment (tsloughter-splunk) [reply CHANGELOG.md:18]: Added.

File: README.md (hunk 1 above)
Comment (tsloughter-splunk) [reply README.md:174]: Added the SDK.

**Pull request merged**